multi-agent/tests/unit/resources/test_resources_service.py

```
@@ -0,0 +1,523 @@
+
+    def test_promote_rejects_cascade_through_disabled_category_dependency(self, service, alice):
+        """Regression test: a dependency in a `builtin_disabled_categories()`
+        category (e.g. a retriever) must reject the whole promotion instead
+        of being silently skipped — leaving the parent public while it still
+        references a resource that end users can never see would defeat the
+        entire cascade-promotion guarantee."""
+        retriever = _make_disabled_category_resource(service, alice)
+        agent = _make_custom_resource(service, alice, name="agent-with-retriever-dep")
+        _link_nested(service, agent, retriever.rid)
+
+        with pytest.raises(ValueError):
```

Ruff (0.16.0)

[warning] `pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)